### DISCUSSION TITLE
Vimeo video loading as full MP4 or segments

### DISCUSSION THREAD

**Discussion by davegoopot** [mozilla/hubs#3117] [General Questions & Discussion] [answered]
We've noticed playing Vimeo videos that some times a full single MP4 file plays and sometimes it is segments. It seems for low-end devices or slow networks the clients have real problems playing the segments version. The same clients typically work fine for a single MP4. What's the details of what is going on for segments vs. full MP4?

Here's a picture showing the segments loading:

![image](https://user-images.githubusercontent.com/3875290/94989866-abfe2380-056f-11eb-965a-213584900bf8.png)

**Comment by davegoopot** [comment, marked as answer]
Also note that the segments seem to randomly restart. This affects all clients at the same time. When the segments restart, the video skips back to the start too. Quite frustrating!

![image](https://user-images.githubusercontent.com/3875290/94991423-00f36700-057b-11eb-9d17-593d3b788fb4.png)

**Reply by davegoopot** [reply]
Figured out what's going on with the videos restarting. If the video is marked as auto-play in Spoke, then it restarts for everyone when a new person enters the Hub. This is a bug I presume? Expected behaviour is surely that auto-play only starts the video if it is not already playing?

**Reply by davegoopot** [reply]
I've raised #3125 for this one.

**Reply by robinkwilson** [reply]
Nice! Thanks for making an issue. 

I'd recommend having the full segment for the mp3 file download for now since that seems to work for you. 

If you'd like to make an issue for the segments, that would help us keep track.
